apps/api/src/test-data/validateTestData.spec.ts

```diff
@@ -19,90 +22,177 @@ export type InMemorySnapshotOfDTOs = {
+
+                    writeFileSync(
+                        testDataFilePath,
+                        JSON.stringify(testDataInDatabaseFormat, null, numberOfSpacesToIndent)
+                    );
                 });
+            });
+        });
+    });
 
-                const testDataInDatabaseFormat =
-                    // Use `collectionNames` not `resourceTypes` as keys
-                    Object.entries(testData).reduce(
-                        (acc, [key, models]) => ({
-                            ...acc,
-                            [getArangoCollectionIDFromResourceType(key as ResourceType)]:
-                                models.map((model) => mapEntityDTOToDatabaseDTO(model)),
-                        }),
-                        {}
+    /**
+    We need to validate our Edge Connection test data
+        For each ResourceType there should be at least one of each of the following
+            an EdgeConnection with a resource of that type as the to member
+            an EdgeConnection with a resource of that type as the from member
+            a (self) EdgeConnection with a resource of that type as the self memeber
+        Every EdgeConnection should satisfy the Edge Connection Invariant Validation
+            i.e. we need to pass the EdgeConnection DTOs through the Edge Connection Validator
+     */
+    describe('test data for edge connections', () => {
+        const { connections: connectionTestData, resources: resourceTestData } = buildTestData();
+
+        const doesMemberWithResourceTypeAndRoleExist = (
+            targetResourceType: ResourceType,
+            targetRole: EdgeConnectionMemberRole
+        ) =>
+            connectionTestData
+                .flatMap(({ members }) => members)
+                .filter(({ compositeIdentifier: { type } }) => type === targetResourceType)
+                .some(({ role }) => role === targetRole);
+
+        Object.values(resourceTypes).forEach((resourceType) => {
+            describe(`the resource type: ${resourceType}`, () => {
+                Object.values(EdgeConnectionMemberRole).forEach((role) => {
+                    it(`should have one instance that is associated with a ${role} connection`, () => {
+                        const result = doesMemberWithResourceTypeAndRoleExist(resourceType, role);
+
+                        expect(result).toBe(true);
+                    });
+                });
+            });
+        });
+
+        connectionTestData.forEach((edgeConnection, index) => {
+            describe(`Edge Connection at index ${index}`, () => {
+                it(`should satisfy the invariants for an Edge Connection`, () => {
+                    const validationResult = edgeConnectionValidator(edgeConnection);
+
+                    expect(validationResult).toBe(Valid);
+                });
+
+                const { members, tagIDs } = edgeConnection;
+
+                it(`should reference only tags that are in the test data`, () => {
```

Comment:
The logic from the next couple blocks should probably eventually make it into the production stream. We are validating required external state for the test data, but we'll also need to do the same when we have commands come in.